pkg/cloudprovider/aws/block_store.go

```diff
@@ -154,20 +178,24 @@ func (b *blockStore) CreateSnapshot(volumeID, volumeAZ string, tags map[string]s
 		return "", errors.WithStack(err)
 	}
 
-	tagsReq := &ec2.CreateTagsInput{}
-	tagsReq.SetResources([]*string{res.SnapshotId})
-
-	ec2Tags := make([]*ec2.Tag, 0, len(tags))
+	tagsReq := &ec2.CreateTagsInput{
+		Resources: []*string{res.SnapshotId},
+	}
 
 	for k, v := range tags {
-		key := k
-		val := v
+		key, val := k, v
```

Comment:
I think you should be able to remove the `key, val` copy and change the contents of this loop to this, if you want:
```go
tagsReq.Tags = append(tagsReq.Tags, (&ec2.Tag{}).SetKey(k).SetValue(v))
```